Begin a universal conversion to f-strings

cmpilato opened the pull request:
As much as possible, use f-strings instead of %-formatting for string interpolation, being careful not to disturb MySQL and date/time formatting placeholders.  Selectively replace some string concatenation with f-strings, as well.

Unrelated drive-by fixes included, also:

- Remove unused `lib/common.py:cmp()` function
- Properly handle `lib/vclib/ccvs/ccvs.py:BlameSource.__init__(..., opt_rev=None)`

cmpilato commented:
I _think_ Copilot and I got 'em all...

futatuki reviewed (commented):
Under reviewing.  I've seen under `bin/` only,  yet.

futatuki reviewed (commented):
I've checked from `lib/accept.py` to `lib/cvsdb.py`.  Still remains from `lib/dbi.py` - ....

futatuki commented:
In lib/dbi.py, `DateTimeFromTicks()`,
```    return "%04d-%02d-%02d %02d:%02d:%02d" % t[:6]```
can be coverted to
```    return "{:04d}-{:02d}-{:02d} {:02d}:{:02d}:{:02d}".format(*t[:6])```
(or unnaturally using f-string :)).  I think it is good, either to replace or not,  however the essence of this PR is standarization, so we should make it clear...

If the arguments for format string is naturally a `dict` or a iterable, we use
(a) old '`%`' operator for readability.
(b) str.format() method with *args or **args.
(c) f-string,  specifing index per place holder.

I like (b).

cmpilato commented on bin/db/loginfo-handler:
```diff
@@ -194,13 +194,13 @@ def CvsNtArgParse(s, repository):
         file_data_list.append(file_data)
 
-        debug('File "%s", old revision %s, new revision %s' % file_data)
+        debug(f'File "{file_data[0]}", old revision {file_data[1]}, new revision {file_data[2]}')
```
Oh, I love your final suggestion!  :trophy:

futatuki commented on lib/viewvc.py:
```diff
@@ -1954,15 +1951,16 @@ def make_time_string(date, cfg):
             if tz < 0:
-                tz = "-%02d:%02d" % (-tz // 3600, (-tz % 3600) // 60)
+                tz = f"{-tz // 3600:02d}:{(-tz % 3600) // 60:02d}"
+                tz = f"-{tz}"
```
Variable name `tz` here is very confusing.  It is used both of for raw timezone offset value (int) and for its string representation (str). 

These 2 lines can be merged:
```                tz = f"-{-tz // 3600:02d}:{(-tz % 3600) // 60:02d}"```

futatuki reviewed (commented):
I've seen files directly under `lib` directory (not seen files under `vcauth` and `vclib` directories).

futatuki commented:
I've seen `lib/vcauth`.  If it is allowed the expression like `root_path = root_path + "/" + "/".join(path_parts)`, nothing to do.

cmpilato commented on lib/viewvc.py:
```diff
@@ -1563,10 +1560,10 @@ def get_time_text(request, interval, num):
     if num == 1:
         attr = text + "_singular"
-        fmt = "%d " + text
+        fmt = f"{{num}} {text}"
     else:
         attr = text + "_plural"
-        fmt = "%d " + text + "s"
+        fmt = f"{{num}} {text}s"
```
For what it's worth, I don't mind using the f-string to do string "connection".  I find it more natural to read that way than mentally construction strings and f-strings literals and string-ish data.

futatuki replied:
I agree that those case here and below are "formatting" rather than "connecting".

futatuki commented on viewvc-install:
```diff
@@ -468,9 +466,7 @@ Just hit [Enter] if a default is okay.
-    print(
-        "Installing ViewVC to %s%s:" % (ROOT_DIR, DESTDIR and " (DESTDIR = %s)" % (DESTDIR) or "")
-    )
+    print(f"Installing ViewVC to {ROOT_DIR}{f' (DESTDIR = {DESTDIR})' if DESTDIR else ''}:")
```
Although I think it is correctly converted from nested % operator into nested f-string and 'and-or' explession into 'if-else' expression, I prefer spritting it into 2 step operations for readability.

futatuki reviewed (commented):
And finally,  I've seen all python code files.

cmpilato commented:
Rebased this on the laster `master`, resolving conflicts in `lib/blame.py` in the obvious way ("delete the file").  I also squashed the commits while I was at it.

futatuki commented:
Please remember add note that there is an extra changes unrelated to the f-string conversion, removing function `cmp` in `lib/common.py`, to commit log.
